### PR TITLE
Prevent freeze app on streaming

### PR DESCRIPTION
**Pull request by kjpolaszek**
It should prevent freeze app on streaming

### CODE REVIEW

File: HysteriaPlayer/HysteriaPlayer.m [hunk 1]
```diff
@@ -305,19 +305,28 @@ - (void)getSourceURLAtIndex:(NSInteger)index preBuffer:(BOOL)preBuffer
...
+            
+            [self setHysteriaIndex:item key:[NSNumber numberWithInteger:index]];
+            
+            if (self.isMemoryCached) {
+                NSMutableArray *playerItems = [NSMutableArray arrayWithArray:self.playerItems];
+                [playerItems addObject:item];
+                self.playerItems = playerItems;
+            }
+            
+            if ([self getLastItemIndex] == index) {
```
Comment (saiday) [inline HysteriaPlayer/HysteriaPlayer.m]: this is a truly needed, much appreciated

**Comment by saiday** [comment]
That is an awesome approach!

**Comment by mkc842** [comment]
What was your intent with this line?

`if ([self getLastItemIndex] == index) {`

For me, it causes the first item in the player to play twice. 

I have commented out the `if` statement, and call `[self insertPlayerItem:item];` unconditionally, as before.

**Comment by patrickjquinn** [comment]
Can we re-open this and find a working solution that doesn't throw errors or repeat the last audio file once another one is added!

**Pull request merged**